internal/k8s/controllers/layer2_status_controller.go

```diff
@@ -61,32 +67,59 @@ func (r *Layer2StatusReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	level.Info(r.Logger).Log("controller", "Layer2StatusReconciler", "start reconcile", req.NamespacedName.String())
 	defer level.Info(r.Logger).Log("controller", "Layer2StatusReconciler", "end reconcile", req.NamespacedName.String())
 
-	svcName := strings.TrimSuffix(req.Name, fmt.Sprintf("-%s", r.NodeName))
-	ipAdvS := r.StatusFetcher(types.NamespacedName{Name: svcName, Namespace: req.Namespace})
+	serviceName, serviceNamespace := req.Name, req.Namespace
 
+	ipAdvS := r.StatusFetcher(types.NamespacedName{Name: serviceName, Namespace: serviceNamespace})
+	var serviceL2statuses v1beta1.ServiceL2StatusList
+	if err := r.Client.List(ctx, &serviceL2statuses, client.MatchingFields{
+		serviceIndexName: types.NamespacedName{Name: serviceName, Namespace: serviceNamespace}.String(),
+	}); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	var errs []error
 	if len(ipAdvS) == 0 {
-		err := r.Client.Delete(ctx, &v1beta1.ServiceL2Status{ObjectMeta: metav1.ObjectMeta{Name: req.Name, Namespace: req.Namespace}})
-		return ctrl.Result{}, client.IgnoreNotFound(err)
+		for key := range serviceL2statuses.Items {
+			if err := r.Client.Delete(ctx, &serviceL2statuses.Items[key]); err != nil && !errors.IsNotFound(err) {
+				errs = append(errs, err)
+			}
```

Comment:
I understand this should probably be handled by the predicate, but let's be on the safe side and validate that the status first belongs to this node (`.Status.Node`) before deleting it.